component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -489,20 +489,39 @@ private void proceedWithOTP(HttpServletResponse response, AuthenticationContext
      */
     private void checkStatusCode(HttpServletResponse response, AuthenticationContext context,
                                  String queryParams, String errorPage) throws AuthenticationFailedException {
+        String screenValue = null;
         boolean isRetryEnabled = SMSOTPUtils.isRetryEnabled(context, getName());
         String loginPage = getLoginPage(context);
         String url = getURL(loginPage, queryParams);
+        String username = String.valueOf(context.getProperty(SMSOTPConstants.USER_NAME));
```

Comment:
Can there be a situation where the username property is null? If so possible NPE

Reply:
In retry follow, there is no chance to get the user name property gets null. It already set in the context when initiate the follow.